presto/dataset.py

```diff
@@ -141,13 +147,111 @@ def __getitem__(self, idx):
         row = self.df.iloc[idx, :]
         eo, mask_per_token, latlon, month, target = self.row_to_arrays(row)
         mask_per_variable = np.repeat(mask_per_token, BAND_EXPANSION, axis=1)
-        num_masked_tokens = sum(sum(mask_per_token))
         return (
             self.normalize_and_mask(eo),
             target,
             np.ones(self.NUM_TIMESTEPS) * (DynamicWorld2020_2021.class_amount),
             latlon,
             month,
-            num_masked_tokens,
             mask_per_variable,
         )
+
+
+class WorldCerealInferenceDataset(Dataset):
+    _NODATAVALUE = 65535
+    Y = "worldcereal_cropland"
+    BAND_MAPPING = {
+        "B02": "B2",
+        "B03": "B3",
+        "B04": "B4",
+        "B05": "B5",
+        "B06": "B6",
+        "B07": "B7",
+        "B08": "B8",
+        # B8A is missing
+        "B11": "B11",
+        "B12": "B12",
+        "VH": "VH",
+        "VV": "VV",
+        "precipitation-flux": "total_precipitation",
+        "temperature-mean": "temperature_2m",
+    }
+
+    def __init__(self):
+        self.path_to_files = data_dir / "inference_areas"
+        self.all_files = list(self.path_to_files.glob("*.nc"))
+
+    def __len__(self):
+        return len(self.all_files)
+
+    @classmethod
+    def nc_to_arrays(
+        cls, filepath: Path
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        ds = cast(xr.Dataset, rioxarray.open_rasterio(filepath, decode_times=False))
+        epsg_coords = ds.rio.crs.to_epsg()
+
+        num_instances = len(ds.x) * len(ds.y)
+        num_timesteps = len(ds.t)
+        eo_data = np.zeros((num_instances, num_timesteps, len(BANDS)))
+        mask = np.zeros((num_instances, num_timesteps, len(BANDS_GROUPS_IDX)))
+        # for now, B8A is missing
```

Comment:
Is it an easy addition to the data? It might be an interesting experiment to see if the model performs much better with it.

However for now I am happy to keep it out so that we still have an apples-to-apples comparison with the CatBoost model.